scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -90,29 +91,65 @@ def create_ngram_index(
                 id = instance.id
                 assert id
                 input_tokens = tokenizer.tokenize(instance.input)
-                for input_ngram in ngrams(input_tokens, n):
-                    if input_ngram not in ngram_index[n]:
-                        ngram_index[n][input_ngram] = set()
-                    ngram_index[n][input_ngram].add(
+                for input_hash_info in get_ngram_hashes(input_tokens, n):
+                    input_hash, start, end = input_hash_info
+                    if input_hash not in ngram_index[n]:
+                        ngram_index[n][input_hash] = set()
+                    ngram_index[n][input_hash].add(
                         EntryDataOverlapKey(stats_key=stats_key, instance_id=id, part=PART_INPUT)
                     )
 
                 # compute reference ngrams
                 for reference in instance.references:
                     reference_unigrams = tokenizer.tokenize(reference)
-                    for reference_ngram in ngrams(reference_unigrams, n):
-                        if reference_ngram not in ngram_index[n]:
-                            ngram_index[n][reference_ngram] = set()
-                        ngram_index[n][reference_ngram].add(
+                    for reference_hash_info in get_ngram_hashes(reference_unigrams, n):
+                        reference_hash, start, end = reference_hash_info
+                        if reference_hash not in ngram_index[n]:
+                            ngram_index[n][reference_hash] = set()
+                        ngram_index[n][reference_hash].add(
                             EntryDataOverlapKey(stats_key=stats_key, instance_id=id, part=PART_REF)
                         )
     return ngram_index
 
 
+def create_hash_to_ngrams(
```

Comment:
Less efficient to separate it out when we do compute this value (since need to hash twice), but much cleaner for code and to enable --skip-check-collision, which will be the common case (i.e. when we aren't testing our hash function)